### ISSUE TITLE
Sec 5.4

### ISSUE THREAD

**Issue by larseggert** [NTAP/rfc8312bis#94]
[Markku Kojo said](https://mailarchive.ietf.org/arch/msg/tcpm/nd4Af8QRQMDkp_R_9ZQMkI4yGAc):
>Sec 5.4
>
>The text correctly states that CUBIC fills queues faster than AIMD TCP and increases the risk of standing queues. Then it proposes queue sizing and AQM as a solution, which is odd. Applying AQM to keep the queues shorter of course decreases the RTT (delay) seen but it does not help with standing queues (they remain standing but are just shorter).

**Comment by bbriscoe** [comment]
The discussion under issue #89 is relevant here.

AQMs help to reduce and in some cases remove the standing queue. An AQM target has to allow for different amplitude sawteeth for different RTTs. In true cubic mode, the brief downward excursions mean that utilization is less sensitive to an AQM target that is lower than the amplitude of the whole sawtooth - it only causes minor underutilization. So the combination of AQMs and true Cubic mode really does help _remove_ standing queues, not just reduce them.

Anyway, the draft doesn't say that AQMs remove standing queues. 

So, I think the draft is correct to point to AQM as a solution to the case of large buffers under 'difficult environments', and the authors might want to add something about the lower sensitivity to under-configured AQM targets, as above.

Nonetheless, I would criticize this section for not really addressing the 'difficult environments' that RFC5033 was talking about: 

- Radio: Cubic is pretty poor on fast adaptation, but still better than Reno. So I don't think we're going to find a reference that praises Cubic in this respect. Therefore, I suggest a sentence admitting Cubic isn't much better than Reno over radio links would be appropriate here. Slides [2](https://bobbriscoe.net/presents/1707ietf/1707tcp-prague-scaling-tensions.pdf#page=2) & [3](https://bobbriscoe.net/presents/1707ietf/1707tcp-prague-scaling-tensions.pdf#page=3) of an iccrg presentation we did summarize the wider problem of the slow adaptation of Cubic in radio environments, but again, other CCs are little better - the next slide moves on to Compound TCP for instance. A reference to [Liu16] would be worthwhile - it's the best paper I've read on the problems TCP CCs have over mobile (and Cubic specifically), and it identifies one major contributor to the problems (receive-window related so not specific to Cubic). 
- Multipath: No issue here relative to Reno AFAICT. No need to mention.
- Tunnels, L2 AQMs, etc: No issue here for a loss-based CC. And any issues with ECN would be no different for Cubic relative to Reno. No need to mention.
- High BDP: Surely this would be the place to call out Cubic's main strength.
- Significantly slow links: No issue here relative to Reno AFAICT (assuming issue #83 is resolved).

[Liu16] K. Liu and J. Y. B. Lee, "On Improving TCP Performance over Mobile Data Networks," IEEE Transactions on Mobile Computing, 2016.

**Comment by larseggert** [comment]
@bbriscoe: would you want to propose a PR that captures your considerations?

**Comment by bbriscoe** [comment]
No, not really. I've given some pointers above that should help. But I need to draw a line between helping vs turning rfc8312bis into my day job (and now becoming my evening job too).

**Comment by larseggert** [comment]
Understood. @lisongxu self-assigned this a while ago, so I'll let him do a PR based on your input.

**Comment by lisongxu** [comment]
Yes. Thanks, @larseggert

**Comment by larseggert** [comment]
@markkukojo, please review #123. I would like to close this issue.

**Comment by markkukojo** [comment]
I'll try to review and answer this and the other pending issues ASAP. Unfortunately, I have very limited cycles for this right now so my apologies for the delay. After this week I hopefully could allocate more cycles.
